packages/wallet/src/redux/sagas/channel-locker.ts

```diff
@@ -0,0 +1,25 @@
+import { LockChannelRequest, channelLocked, channelUnlocked } from '../actions';
+import { take, takeEvery, put, call } from 'redux-saga/effects';
+
+export function* channelLocker() {
+  yield takeEvery('WALLET.LOCKING.LOCK_CHANNEL_REQUEST', lockChannel);
+}
+
+function* lockChannel(lockRequest: LockChannelRequest) {
```

Comment:
This is the saga responsible for locking a channel with `web-locks`. Right now it requires that any protocol that wants to (possibly) modify a channel dispatches a `LOCK_CHANNEL_REQUEST` and waits until it receives a `CHANNEL_LOCKED` action. Currently not implemented is proper handling of when a channel is already locked.